meta-mender-core/classes/mender-helpers.bbclass

```diff
@@ -19,6 +22,9 @@ get_uboot_device_from_device() {
         /dev/mmcblk[0-9])
             echo ${1#/dev/mmcblk}
             ;;
+        ubi[0-9])
+            echo ubi
```

Comment:
Alright, there doesn't seem to be any option here which isn't at least a little confusing, so I'll try to make a suggestion which is as non-confusing and consistent as possible, hopefully also addressing your other lengthy comment:

* I suppose since a specification without `/dev` is so common, it makes sense to have `MENDER_STORAGE_DEVICE` refer to `ubi0` directly.

* Rename `MENDER_UBOOT_STORAGE_DEVICE` to `MENDER_UBOOT_STORAGE_DEVICE_NUMBER`, not just for ubi, but for existing sdimg as well. A number is what it has been all along, so this is more accurate and descriptive.

* Similarly, rename `get_uboot_device_from_device` to `get_uboot_device_number_from_device` and have it return `${1#ubi}`, IOW just a number.

* Taken together, the previous two changes will make `mender_uboot_dev` into a number. Now replace:
  ```
  ubi part ${mender_uboot_dev}
  ```
  with:
  ```
  ubi part ${mender_uboot_if}${mender_uboot_dev}
  ```

* For Mender opening the ubi block device, we got two choices:

  1. Add the reference `/dev/` when rendering the config file in mender_0.1.bb.

  2. Have the Mender client itself accept bare references to `ubi*`, and transparently add the right path in the code. I can help with this, if you need it.

  I think I vote for the second option, since the Mender client appears to be the only component in this puzzle that isn't capable of handling bare ubi references.

* Create the U-Boot variable `mender_uboot_root_name` as an alias to `mender_kernel_root_name`. This has no functional impact, it's just to have unambiguous variables for uboot and kernel references, respectively.

* Replace:
  ```
  ubifsmount ${mender_kernel_root_name}
  ```
  with:
  ```
  ubifsmount ${mender_uboot_root_name}
  ```

What do you think?